src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts

```diff
@@ -241,12 +245,16 @@ export class CreateScenariosComponent implements OnInit {
           return NEVER;
         })
       )
-      .subscribe(() => {
+      .subscribe((newScenario) => {
+        // Setting the new scenario id
+        this.scenarioId = newScenario.id;
+        this.scenarioName = newScenario.name
```

Comment:
are you using `this.scenarioName ` anywhere else? Might as well just do  
```
this.planStateService.updateStateWithScenario(
      this.scenarioId || null,
      this.scenarioName || null
    );
```

directly? not a blocker though just a suggestion

Reply:
Updated, let me know if you find something else